AutoRouteManager.php

```diff
@@ -64,10 +71,30 @@ public function __construct(
      */
     public function buildUriContextCollection(UriContextCollection $uriContextCollection)
     {
-        $this->getUriContextsForDocument($uriContextCollection);
+        $this->collectionBuilder->build($uriContextCollection);
 
         foreach ($uriContextCollection->getUriContexts() as $uriContext) {
-            $existingRoute = $this->adapter->findRouteForUri($uriContext->getUri(), $uriContext);
+            $subjectObject = $uriContextCollection->getSubjectObject();
+
+            if (null !== $uriContext->getLocale()) {
+                $subjectObject = $this->adapter->translateObject($uriContextCollection->getSubjectObject(), $uriContext->getLocale());
```

Comment:
Have moved object translation back to the Manager (as it needed to translate the objects before calling the UriGenerator).